examples/experimental/slpk-in-browser/src/custom-tile-3d-layer.ts

```diff
@@ -0,0 +1,41 @@
+import {Tile3DLayer, Tile3DLayerProps} from '@deck.gl/geo-layers';
+import {UpdateParameters} from '@deck.gl/core/typed';
+
+export default class CustomTile3DLayer<
+  DataT = any,
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  ExtraProps extends {} = {}
+> extends Tile3DLayer<DataT, Tile3DLayerProps & ExtraProps> {
+  static layerName = 'CustomLayer';
+  static defaultProps = Tile3DLayer.defaultProps as any;
+
+  //@ts-expect-error
```

Comment:
This is a lot of expect-errors can we do better? I believe deck.gl source has a trick for typing state, as one example

Reply:
The most members of deck.gl layers are `private` that doesn't allow to inherit them. Probably, we could make members of deck.gl layers `protected`. It will make them more flexible for inheritance.